packages/oui-action-menu/src/action-menu-item.controller.js

```diff
@@ -0,0 +1,54 @@
+import { addBooleanParameter } from '@oui-angular/common/component-utils'
+
+const baseClass = 'oui-action-menu__item'
+const dividerClass = 'oui-action-menu__item_divider'
+const buttonClasses = 'oui-button oui-button_action-menu'
+
+const linkTemplate = (dividerClass, external) => `
+  <li class="${baseClass} ${dividerClass}">
+    <a
+      class="${buttonClasses}"
+      data-ng-href="{{::$ctrl.href}}"${external ? ' target="_blank" rel="noopener"' : ''}>
+      <span data-ng-bind="::$ctrl.text"></span>
+      ${external ? '<i class="oui-icon oui-icon-external_link" aria-hidden="true"></i>' : ''}
+    </a>
+  </li>`
+
+const buttonTemplate = dividerClass => `
+  <li class="${baseClass} ${dividerClass}">
+    <button
+      class="${buttonClasses}"
+      type="button"
+      data-ng-click="$ctrl.onClick()"
+      data-ng-bind="::$ctrl.text"></button>
+  </li>`
+
+export default class {
+  constructor ($attrs, $compile, $element, $scope) {
+    'ngInject'
+
+    this.$attrs = $attrs
+    this.$compile = $compile
+    this.$element = $element
+    this.$scope = $scope
+  }
+
+  $onInit () {
+    addBooleanParameter(this, 'divider')
+    addBooleanParameter(this, 'external')
+  }
+
+  $postLink () {
+    let compiled
+
+    const dividerClassName = this.divider ? dividerClass : ''
+
+    if (this.$attrs.onClick) {
+      compiled = this.$compile(buttonTemplate(dividerClassName))
+    } else {
+      compiled = this.$compile(linkTemplate(dividerClassName, this.external))
+    }
```

Comment:
Why not use 2 different directives ?
I know it's to keep only one directive for both case, but we'll have a simpler code with `action-menu-item-link` and `action-menu-item-button` (for example). And we don't have to check the bindings.

It applies to https://github.com/ovh-ux/ovh-ui-angular/pull/17 too. 
Maybe the test error come from this kind of unnecessary complexity.

Reply:
The complexity is to create 2 different components for a very similar feature I think. Because what prevent us to think about a `oui-action-menu-item-link-external` then?
`$compile` exists to be used.

Reply:
We can say that for a lot of functions/methods.
eg. `ngIf` or `ngSwitch` in the template (with one-time binding) -> Simpler